[Bug] [Source-Http] Invalid mime type when sending POST requets with json body

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

Caused by: org.apache.seatunnel.connectors.seatunnel.http.exception.HttpConnectorException: ErrorCode:[HTTP-02], ErrorDescription:[The request is failed] - http client execute exception, http response status code:[415], content:[{"timestamp":"2025-05-26T07:08:08.588+00:00","status":415,"error":"Unsupported Media Type","exception":"org.springframework.web.HttpMediaTypeNotSupportedException","message":"Invalid mime type \"application/json,application/json\": Invalid token character ',' in token \"json,application/json\"","path":"/api/demo/lkjt/getContracts"}]


### SeaTunnel Version

2.3.10

### SeaTunnel Config

```conf
{
  "env":{
    "job.mode":"BATCH",
    "pipeline.name":"job_health_day",
    "execution.parallelism":"1"  },
  "source" : [
    {
      "schema" : {
        "fields" : {
          "contract_amount" : "String",
          "handler" : "String",
          "contract_id" : "String",
          "contract_status" : "String",
          "contract_name" : "String",
          "project_name" : "String",
          "payment_type" : "String",
          "party_b" : "String",
          "contract_date" : "String",
          "party_a" : "String",
          "party_c" : "String",
          "paid_amount" : "String",
          "department" : "String",
          "contract_origin" : "String"
        }
      },
      "url" : "https://url.com/api/demo/lkjt/getContracts",
      "method" : "POST",
      "format" : "json",
      "body" : "{\"params\":{\"conditions\":{\"region\":\"EMEA\",\"userType\":\"RETURN\"},\"columns\":{\"startDate\":\"2023-10-01\",\"endDate\":\"2023-11-05\"},\"variables\":{\"hgssiteid\":\"500024371\"}}}",
      "headers" : {
        "authorization" : "Bearer token"
      },
      "plugin_name" : "Http",
      "result_table_name" : null,
      "content_field" : "$.data.*"
    }
  ],
  "sink":[
    {
      "plugin_name":"Console"
    }
  ]
}
```

### Running Command

```shell
org.apache.seatunnel.example.flink.v2.SeaTunnelApiExample
```

### Error Exception

```log
Caused by: org.apache.seatunnel.connectors.seatunnel.http.exception.HttpConnectorException: ErrorCode:[HTTP-02], ErrorDescription:[The request is failed] - http client execute exception, http response status code:[415], content:[{"timestamp":"2025-05-26T07:08:08.588+00:00","status":415,"error":"Unsupported Media Type","exception":"org.springframework.web.HttpMediaTypeNotSupportedException","message":"Invalid mime type \"application/json,application/json\": Invalid token character ',' in token \"json,application/json\"","path":"/api/demo/lkjt/getContracts"}]
	at org.apache.seatunnel.connectors.seatunnel.http.source.HttpSourceReader.pollAndCollectData(HttpSourceReader.java:150)
	at org.apache.seatunnel.connectors.seatunnel.http.source.HttpSourceReader.internalPollNext(HttpSourceReader.java:362)
	at org.apache.seatunnel.connectors.seatunnel.http.source.HttpSourceReader.pollNext(HttpSourceReader.java:330)
	at org.apache.seatunnel.translation.flink.source.FlinkSourceReader.pollNext(FlinkSourceReader.java:99)
```

### Zeta or Flink or Spark Version

flink 1.18

### Java or Scala Version

java 11

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProvider.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProvider.java
@@ -465,7 +465,7 @@ public class HttpClientProvider implements AutoCloseable {
         headers.forEach(request::addHeader);
     }
 
-    private void addBody(HttpEntityEnclosingRequestBase request, Map<String, Object> body)
+    static void addBody(HttpEntityEnclosingRequestBase request, Map<String, Object> body)
             throws UnsupportedEncodingException {
         if (MapUtils.isEmpty(body)) {
             body = new HashMap<>();
@@ -489,10 +489,8 @@ public class HttpClientProvider implements AutoCloseable {
                 request.setEntity(new UrlEncodedFormEntity(parameters, ENCODING));
             }
         } else {
-            request.addHeader(HTTP.CONTENT_TYPE, APPLICATION_JSON);
             StringEntity entity =
                     new StringEntity(JsonUtils.toJsonString(body), ContentType.APPLICATION_JSON);
-            entity.setContentEncoding(new BasicHeader(HTTP.CONTENT_TYPE, APPLICATION_JSON));
             request.setEntity(entity);
         }
     }

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
@@ -31,11 +31,22 @@ class HttpClientProviderTest {
     void addBody() throws Exception {
         HttpPost post = new HttpPost("http://localhost:8080");
         Map<String, Object> body = new HashMap<>();
-        body.put("key1", "value1");
         Header[] originalHeaders = post.getAllHeaders();
 
         HttpClientProvider.addBody(post, body);
-        Assertions.assertEquals(post.getAllHeaders(), originalHeaders);
+
+        Header[] currentHeaders = post.getAllHeaders();
+        Assertions.assertEquals(originalHeaders.length, currentHeaders.length);
+        for (int i = 0; i < originalHeaders.length; i++) {
+            Assertions.assertEquals(
+                    originalHeaders[i].getName(),
+                    currentHeaders[i].getName(),
+                    "Header name mismatch at index " + i);
+            Assertions.assertEquals(
+                    originalHeaders[i].getValue(),
+                    currentHeaders[i].getValue(),
+                    "Header value mismatch at index " + i);
+        }
         Assertions.assertNull(post.getEntity().getContentEncoding());
     }
 }

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
@@ -35,6 +35,7 @@ class HttpClientProviderTest {
 
         HttpClientProvider.addBody(post, body);
 
+        // ensure the original headers are preserved
         Header[] currentHeaders = post.getAllHeaders();
         Assertions.assertEquals(originalHeaders.length, currentHeaders.length);
         for (int i = 0; i < originalHeaders.length; i++) {
@@ -47,6 +48,7 @@ class HttpClientProviderTest {
                     currentHeaders[i].getValue(),
                     "Header value mismatch at index " + i);
         }
+        // ensure no manually set content type or encoding
         Assertions.assertNull(post.getEntity().getContentEncoding());
     }
 }

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.connectors.seatunnel.http.client;
 import org.apache.http.Header;
 import org.apache.http.client.methods.HttpPost;
 
-import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 import java.util.HashMap;
@@ -32,10 +32,10 @@ class HttpClientProviderTest {
         HttpPost post = new HttpPost("http://localhost:8080");
         Map<String, Object> body = new HashMap<>();
         body.put("key1", "value1");
-        Header[] orignalheaders = post.getAllHeaders();
+        Header[] originalHeaders = post.getAllHeaders();
 
         HttpClientProvider.addBody(post, body);
-        Assert.assertEquals(post.getAllHeaders(), orignalheaders);
-        Assert.assertNull(post.getEntity().getContentEncoding());
+        Assertions.assertEquals(post.getAllHeaders(), originalHeaders);
+        Assertions.assertNull(post.getEntity().getContentEncoding());
     }
 }

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
@@ -1,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seatunnel.connectors.seatunnel.http.client;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.HttpPost;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class HttpClientProviderTest {
+
+    @Test
+    void addBody() throws Exception {
+        HttpPost post = new HttpPost("http://localhost:8080");
+        Map<String, Object> body = new HashMap<>();
+        body.put("key1", "value1");
+        Header[] orignalheaders = post.getAllHeaders();
+
+        HttpClientProvider.addBody(post, body);
+        Assert.assertEquals(post.getAllHeaders(), orignalheaders);
+        Assert.assertNull(post.getEntity().getContentEncoding());
+    }
+}

--- a/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
+++ b/seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProviderTest.java
@@ -32,7 +32,6 @@ class HttpClientProviderTest {
         HttpPost post = new HttpPost("http://localhost:8080");
         Map<String, Object> body = new HashMap<>();
         Header[] originalHeaders = post.getAllHeaders();
-
         HttpClientProvider.addBody(post, body);
 
         // ensure the original headers are preserved

